### PR TITLE
Fix: Enhance datasource httpd include upcase

### PR DESCRIPTION
**Pull request by wushiqinlou**
Signed-off-by: jiazhang <jiazhang@redhat.com>

rh-pre-commit.version: 2.3.1
rh-pre-commit.check-secrets: ENABLED

### All Pull Requests:

Check all that apply:

* [x] Have you followed the guidelines in our Contributing document, including the instructions about commit messages?
* [x] No Sensitive Data in this change?
* [x] Is this PR to correct an issue?
* [ ] Is this PR an enhancement?

### Complete Description of Additions/Changes:
lower cases of "Include"/"IncludeOptional" can also work. Current code ignores the case like "include <file path>". This PR can fix this issue

### CODE REVIEW

File: insights/specs/datasources/httpd.py [hunk 1]
```diff
@@ -78,7 +78,7 @@ def _get_all_include_conf(root, glob_path):
                             section_number = section_number - 1
                         elif line.startswith("<"):
                             section_number = section_number + 1
-                        if section_number == 0 and (line.startswith("Include ") or line.startswith("IncludeOptional ")):
+                        if section_number == 0 and (line.lower().startswith("include ") or line.lower().startswith("includeoptional ")):
```
Comment (JoySnow) [inline insights/specs/datasources/httpd.py:81]: @wushiqinlou As this line also mainly check for `line.startswith()`, what about change this `if` to `elif`?

File: insights/specs/datasources/httpd.py [hunk 2]
```diff
@@ -107,7 +107,7 @@ def get_httpd_configuration_files(httpd_root):
                     section_number = section_number + 1
                 if line.startswith("ServerRoot "):
```
Comment (JoySnow) [inline insights/specs/datasources/httpd.py:108]: Same case here, what about change this `if` to `elif`?

File: insights/specs/datasources/httpd.py (hunk 1 above)
Comment (wushiqinlou) [reply insights/specs/datasources/httpd.py]: @JoySnow - updated, please check

File: insights/specs/datasources/httpd.py (hunk 2 above)
Comment (wushiqinlou) [reply insights/specs/datasources/httpd.py]: @JoySnow - updated, please check

**Review by JoySnow — APPROVED** [review]
@wushiqinlou Thanks for the update! 👍

File: insights/specs/datasources/httpd.py [hunk 3]
```diff
@@ -78,7 +78,7 @@ def _get_all_include_conf(root, glob_path):
                             section_number = section_number - 1
                         elif line.startswith("<"):
                             section_number = section_number + 1
-                        if section_number == 0 and (line.startswith("Include ") or line.startswith("IncludeOptional ")):
+                        elif section_number == 0 and (line.lower().startswith("include ") or line.lower().startswith("includeoptional ")):
```
Comment (xiangce) [inline insights/specs/datasources/httpd.py:81]: with the `line.lower()` being introduced, a tiny computing will be done before the two `startswith`, in this case, better change to one `startswith` as 
```
line.lower().startswith(("include ", "includeoptional "))
```

File: insights/specs/datasources/httpd.py [hunk 4]
```diff
@@ -105,9 +105,9 @@ def get_httpd_configuration_files(httpd_root):
                     section_number = section_number - 1
                 elif line.startswith("<"):
                     section_number = section_number + 1
-                if line.startswith("ServerRoot "):
+                elif line.startswith("ServerRoot "):
                     server_root = line.strip().split()[-1].strip().strip('"\'')
-                elif section_number == 0 and (line.startswith("Include ") or line.startswith("IncludeOptional ")):
+                elif section_number == 0 and (line.lower().startswith("include ") or line.lower().startswith("includeoptional ")):
```
Comment (xiangce) [inline insights/specs/datasources/httpd.py:110]: the same as above

File: insights/specs/datasources/httpd.py [hunk 5]
```diff
@@ -105,9 +105,9 @@ def get_httpd_configuration_files(httpd_root):
                     section_number = section_number - 1
                 elif line.startswith("<"):
                     section_number = section_number + 1
-                if line.startswith("ServerRoot "):
+                elif line.startswith("ServerRoot "):
```
Comment (xiangce) [inline insights/specs/datasources/httpd.py:108]: will this be `line.lower().startswith("serverroot ")`?

**Review by xiangce — APPROVED** [review]

**Pull request merged**